### PR TITLE
feat(DFPL-83): Sending e-mails or letters with Placement Order or Pla…

### PR DESCRIPTION
**Pull request by jvictorcf**
…cement order notification

**Before creating a pull request make sure that:**

- [ ] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by tomaszpowroznik — DISMISSED** [review]

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/IntegrationTestConstants.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.fpl.testingsupport;
+
+public interface IntegrationTestConstants {
```
Comment (tomaszpowroznik) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/IntegrationTestConstants.java:3]: there is already uk.gov.hmcts.reform.fpl.Constants for things like this

**Review by jacmurphy — DISMISSED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/GeneratedPlacementOrderEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            .filter(noticeDoc -> PARENT_TYPES.contains(noticeDoc.getType()))
+            .map(PlacementNoticeDocument::getRespondentId)
+            .map(respondentId -> caseData.findRespondent(respondentId).orElseThrow())
+            .map(Element::getValue)
+            .collect(Collectors.toList());
+
+        Set<String> emailRecipients = new HashSet<>();
+        List<Recipient> postRecipients = new ArrayList<>();
+        for (Respondent parent : parents) {
+            boolean isParentRepresented = Objects.isNull(parent.getSolicitor());
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/GeneratedPlacementOrderEventHandler.java:88]: isParentNotRepresented

File: service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/GeneratedPlacementOrderEventHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        verify(notificationService).sendEmail(
+            PLACEMENT_ORDER_GENERATED_NOTIFICATION_TEMPLATE,
+            Set.of(FATHER_SOLICITOR_EMAIL, MOTHER_SOLICITOR_EMAIL, PRIVATE_SOLICITOR_USER_EMAIL),
+            notifyData,
+            TEST_CASE_ID
+        );
+    }
+
+    @Test
+    void shouldSendOrderNotification_ToUnrepresentedParentsByPost_And_NotToUnrepresentedChildSolicitorByEmail() {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/GeneratedPlacementOrderEventHandlerTest.java:178]: And_**Not**ToUnrepresentedChildSolicitorByEmail -> And_ToUnrepresentedChildSolicitorByEmail

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/IntegrationTestConstants.java (hunk 1 above)
Comment (jvictorcf) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/IntegrationTestConstants.java:3]: I wanted to make this accessible only in Integration test.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/GeneratedPlacementOrderEventHandlerTest.java (hunk 3 above)
Comment (jvictorcf) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/GeneratedPlacementOrderEventHandlerTest.java:178]: It's actually correct. Unrepresented children won't get a message.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/IntegrationTestConstants.java (hunk 1 above)
Comment (jvictorcf) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/IntegrationTestConstants.java:3]: as this is only applicable in that context's configuration

**Review by tomaszpowroznik — APPROVED** [review]

**Pull request merged**